DS and Algorithm/Day3/Selectionsort.py

```diff
@@ -0,0 +1,40 @@
+def selection_sort(arr):
+    """
+    Sorts an array using the Selection Sort algorithm.
+
+    Parameters:
+    - arr (list): The input list to be sorted in-place.
+
+    Returns:
+    None (The input list is sorted in-place).
+    """
+
+    # Get the length of the array
+    n = len(arr)
+
+    # Traverse the entire array
+    for i in range(n):
+        # Assume the current index is the minimum
+        min_index = i
+
+        # Iterate through the unsorted part of the array to find the minimum element
+        for j in range(i + 1, n):
+            # Compare the current element with the assumed minimum
+            if arr[j] < arr[min_index]:
+                # If the current element is smaller, update the minimum index
+                min_index = j
+
+        # Swap the found minimum element with the first element in the unsorted part
+        arr[i], arr[min_index] = arr[min_index], arr[i]
```

Comment:
**Performance Issue**: Selection sort has a time complexity of O(n^2), making it inefficient for large datasets compared to more advanced sorting algorithms like quicksort or mergesort. <br> **Fix**: Replace the selection sort algorithm with a more efficient sorting algorithm, such as quicksort or mergesort, to improve the overall performance, especially for large arrays. <br> **Code Suggestion**: 
 ```
 def quicksort(arr):
     if len(arr) <= 1:
         return arr
     pivot = arr[len(arr) // 2]
     left = [x for x in arr if x < pivot]
     middle = [x for x in arr if x == pivot]
     right = [x for x in arr if x > pivot]
     return quicksort(left) + middle + quicksort(right)

 arr = [64, 34, 25, 12, 22, 11, 90]
 arr = quicksort(arr)
 print("Sorted array:", arr)
 ```